.github/workflows/validation.yaml

```diff
@@ -94,3 +104,30 @@ jobs:
       - name: Run lint script
         run: npm run lint
         shell: bash
+
+  run-tests:
+    name: Run optional tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          token: ${{ secrets.GIT_TOKEN_BASIC || github.token }}
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+
+      - name: Install node dependencies
+        run: npm ci
+        shell: bash
+
+      - name: Run unit tests
+        if: inputs.runUnitTests == true
+        run: npm run test
+        shell: bash
+
+      - name: Run e2e tests
+        if: inputs.runE2eTests == true
+        run: npm run e2e
+        shell: bash
```

Comment:
Any thoughts on using an array instead of booleans. Then using matrix/strategy and just `npm run <item>`. Have a default value of 0 items or make the default `test` to "encourage" folks to have a test step :). I ain't married to one way or the other, the array just seems more extensible without adding much complexity.

Reply:
Making the default to any sort of "test" ~will~ may break any workflow that's already using `validation.yaml`

Reply:
/me thinks about matrix.

Reply:
Let's wait for @bmoseley-armor 's opinion

Reply:
@straineanu As long as you're thinking of using a strategy to do `npm run <item>`, why not using the strategy to just do `<item>`?

Then we can even use a strategy like `pipenv run test`. Or `mvn test`. Or whatever you guys use for .Net.

Reply:
@straineanu Using a Strategy matrix will create separate jobs. This means that each job in the matrix might be on different runners. That in turn means I'd have to 

1. Checkout the repo
2. Setup NodeJS
3. npm ci

for each job in the matrix. 

_Are we ok with that?_ Or do we prefer to do the above steps once, and then run `test` and `e2e` depending on the conditionals?

Reply:
Good call re: strategy at the job level. I'd forgotten there isn't a step version of strategy/matrix. I don't know about best practice, but I agree that in the npm use case a strategy matrix would just make noisy output for a very small benefit. I rescind my suggestion of strategy matrix, sorry about that.

I haven't used github actions enough yet to get a good feel on passing in an array versus a bunch of individual booleans. I'm still inclined to think an array is more extensible/flexible, but I'm not convinced its worth the complexity yet. I'm just excited for a github pr conversation.